### PR TITLE
ME0Muon bug fix for 81x

### PR DESCRIPTION
**Pull request by jshlee**
this bug stopped ME0Muon being made in trackerMuon collection
@calabria @kpedro88

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc [hunk 1]
```diff
@@ -804,9 +804,9 @@ void TrackDetectorAssociator::fillMuon( const edm::Event& iEvent,
                  }
               }
 	}
-	// GEM Chamber   
-	else if (parameters.useGEM){
-	  if(const GEMSuperChamber* chamber = dynamic_cast<const GEMSuperChamber*>(geomDet) ) {	 
+	// GEM Chamber or ME0 Chamber
+	else if (parameters.useGEM || parameters.useME0){
```
Comment (cvuosalo) [inline TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc]: This change seems to suggest that `useME0` can be false even though `geomDet` is an ME0 chamber. Why would that be so?

Comment (slava77) [inline TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc]: looks like checks for values of useGEM and useME0 should be added to individual GEM and ME0 chamber ifs

Comment (cvuosalo) [inline TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc]: Yes, and line 808 should just be:
```
else {
```

Comment (jshlee) [inline TrackingTools/TrackAssociator/src/TrackDetectorAssociator.cc]: @cvuosalo - I changed to your suggestion.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@jshlee: Please fix this existing problem found by the static analyzer:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16032/15465/llvm-analysis/report-389cfe.html#EndPath
`std::isnan` needs to be changed to `edm::isNotFinite`.

**Comment by jshlee** [comment]
@cvuosalo - done

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
@cvuosalo, @slava77, @davidlange6 ping for pre13

**Comment by cvuosalo** [comment]
urgent

**Comment by cvuosalo** [comment]
@jshlee, @kpedro88: Do we actually want to enable ME0 muons for the tracking muon iteration as this code does? Previously, they were masked by the bug, but do we really want them included in this tracking step?

**Comment by jshlee** [comment]
@cvuosalo: yes we would like this. Currently, ME0Muon is run as a standalone collection and will be replaced by this later on.

**Comment by slava77** [comment]
@calabria @jshlee 
the question was about using ME0 muons for the iterative tracking (RecoMuon/MuonIdentification/python/earlyMuons_cfi.py )
Are the ME0 tracker muons good enough to seed iterative tracking?
If not, it would be prudent to disable them in the early muon reco and leave in the standard muon "1stStep" reconstruction.

**Comment by slava77** [comment]
I found an answer to the same question:
https://github.com/cms-sw/cmssw/pull/14833#issuecomment-232278585
``` The idea was to go fully integrated already in the muon-driven tracking iterations, but not for identification. ```
So, we are good to go then, unless that decision has changed

**Comment by cvuosalo** [comment]
+1

For #16032 9874e7aae4d7f9c6ff1dee729531cf59c31831e8

Bug fix to include ME0 muons in the tracker muon collection.

The code changes are satisfactory. Jenkins tests against baseline CMSSW_8_1_X_2016-10-06-2300 show the expected differences in Phase 2 workflows as formerly missing ME0 muons are included and affect downstream quantities.

**Comment by davidlange6** [comment]
+1

**Pull request merged**